flytekit/extend/backend/base_agent.py

```diff
@@ -192,12 +206,17 @@ def execute(self, **kwargs) -> typing.Any:
                 progress.start_task(task)
                 time.sleep(1)
                 if agent.asynchronous:
-                    res = asyncio.run(agent.async_get(dummy_context, metadata))
+                    res = loop.run_until_complete(agent.async_get(dummy_context, metadata))
                 else:
                     res = agent.get(dummy_context, metadata)
                 state = res.resource.state
                 logger.info(f"Task state: {state}")
 
+        if agent.asynchronous:
+            loop.close()
+            if self.is_canceled:
+                sys.exit(1)
```

Comment:
we should put these codes in the while loop since `loop.run_until_complete(agent.async_get(dummy_context, metadata))` may always return `running` when using the sensor.

Reply:
Done, please review, thanks a lot!